grazel-gradle-plugin/src/test/kotlin/com/grab/grazel/migrate/android/AndroidInstrumentationBinaryDataExtratorTest.kt

```diff
@@ -0,0 +1,160 @@
+import com.grab.grazel.gradle.DefaultRepositoryDataSource
+import com.grab.grazel.gradle.FakeAndroidVariantsExtractor
+import com.grab.grazel.gradle.KOTLIN_ANDROID_PLUGIN
+import com.grab.grazel.gradle.dependencies.ArtifactsConfig
+import com.grab.grazel.gradle.dependencies.DefaultDependenciesDataSource
+import com.grab.grazel.gradle.dependencies.DefaultDependencyResolutionService
+import com.grab.grazel.gradle.dependencies.GradleDependencyToBazelDependency
+import com.grab.grazel.util.doEvaluate
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.the
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+
+class AndroidInstrumentationBinaryDataExtratorTest : GrazelPluginTest() {
+    private lateinit var rootProject: Project
+    private lateinit var subProject: Project
+    private lateinit var subProjectDir: File
+
+    private lateinit var androidInstrumentationBinaryDataExtractor: AndroidInstrumentationBinaryDataExtractor
+    private lateinit var androidVariantsExtractor: FakeAndroidVariantsExtractor
+    private lateinit var gradleDependencyToBazelDependency: GradleDependencyToBazelDependency
+
+    @get:Rule
+    val temporaryFolder = TemporaryFolder()
+
+    @Before
+    fun setup() {
+        val rootProjectDir = temporaryFolder.newFolder("project")
+        rootProject = buildProject("root", projectDir = rootProjectDir)
+
+        subProjectDir = File(rootProjectDir, "subproject").apply {
+            mkdirs()
+        }
+        subProject = buildProject("subproject", rootProject)
+        subProject.run {
+            plugins.apply {
+                apply(ANDROID_LIBRARY_PLUGIN)
+                apply(KOTLIN_ANDROID_PLUGIN)
+            }
+            extensions.configure<LibraryExtension> {
+                defaultConfig {
+                    compileSdk = 30
+                    testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+                }
+            }
+        }
+        androidVariantsExtractor = FakeAndroidVariantsExtractor()
+        gradleDependencyToBazelDependency = GradleDependencyToBazelDependency()
+        File(subProjectDir, "src/main/AndroidManifest.xml").apply {
+            parentFile.mkdirs()
+            createNewFile()
+            writeText(
+                """
+                    <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+                        package="com.example.androidlibrary">
+                    </manifest>
+                """.trimIndent()
+            )
+        }
+
+        val variantDataSource = DefaultAndroidVariantDataSource(DefaultAndroidVariantsExtractor())
+        val configurationDataSource = DefaultConfigurationDataSource(variantDataSource)
+        val repositoryDataSource = DefaultRepositoryDataSource(rootProject)
+
+        val dependenciesDataSource = DefaultDependenciesDataSource(
```

Comment:
Instead of constructing this manually, you could use Dagger to construct it for you by exposing it in GrazelComponent and then calling `rootProject.createGrazelComponent` as done here 

https://github.com/grab/Grazel/blob/bbf52e11261d412d43385371cfb7b5aa32655ebc/grazel-gradle-plugin/src/test/kotlin/com/grab/grazel/migrate/AndroidWorkspaceRepositoriesTest.kt#L40

Reply:
Thanks! I've updated the commit to using `GrazelComponent `. Now the test codes are much cleaner and easier to read